usaspending_api/awards/v2/filters/transaction.py

```diff
@@ -0,0 +1,286 @@
+from usaspending_api.awards.models import Transaction
+from usaspending_api.common.exceptions import InvalidParameterException
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+# TODO: Performance when multiple false values are initially provided
+def transaction_filter(filters):
+
+    queryset = Transaction.objects.all()
+    for key, value in filters.items():
+        # check for valid key
+        if value is None:
+            raise InvalidParameterException('Invalid filter: ' + key + ' has null as its value.')
+
+        key_list = ['keyword',
+                    'time_period',
+                    'award_type_codes',
+                    'agencies',
+                    'legal_entities',
+                    'recipient_scope',
+                    'recipient_locations',
+                    'recipient_type_names',
+                    'place_of_performance_scope',
+                    'place_of_performance_locations',
+                    'award_amounts',
+                    'award_ids',
+                    'program_numbers',
+                    'naics_codes',
+                    'psc_codes',
+                    'contract_pricing_type_codes',
+                    'set_aside_type_codes',
+                    'extent_competed_type_codes']
+
+        if key not in key_list:
```

Comment:
This one is checking against a key list while `awards` is not. Can we make them consistent with each other?

Reply:
awards is actually completely overhauled in spending_by_geography.

Reply:
Isn't `spending_by_geography` also based on Transactions also?